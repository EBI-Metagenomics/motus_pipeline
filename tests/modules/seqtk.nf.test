nextflow_process {

    name "Test module seqtk"
    script "modules/seqtk.nf"
    process "SEQTK"

    test("seqtk fastq to fasta") {

        when {
            params {
                name = "mgtest_merged"
                reads = "${baseDir}/tests/modules/fixtures/bwa/"
            }
            process {
                """
                input[0] = channel.fromPath("${params.reads}/${params.name}.fastq.gz", checkIfExists: true)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}