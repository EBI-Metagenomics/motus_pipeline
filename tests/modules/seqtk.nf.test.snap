{
    "seqtk fastq to fasta": {
        "content": [
            {
                "sequence": [
                    "mgtest_merged.fasta:md5,c9f9372244986bfecaba2464ecf8feba"
                ]
            }
        ],
        "timestamp": "2023-03-14T15:32:44+0000"
    }
}